### PR TITLE
imxrt-multi: Implement i2c driver

### PR DESCRIPTION
**Pull request by gerard5**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (refactoring, style fixes, git/CI config, submodule management, no code logic changes)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: multi/imxrt-multi/imxrt-multi.c [hunk 1]
```diff
@@ -410,6 +413,23 @@ static int createDevFiles(void)
...
+
+#ifdef __CPU_IMXRT117X
+
+#if I2C5
+	if (mkFile(&dir, id_i2c5, "i2c5", multi_port) < 0) {
+		return -1;
+	}
+#endif
+
+#if I2C4
```
Comment (Darchiv) [inline multi/imxrt-multi/imxrt-multi.c:425]:
```suggestion
#if I2C6
```

File: multi/imxrt-multi/imxrt-multi.c [hunk 2]
```diff
@@ -419,9 +439,6 @@ static int createDevFiles(void)
 	if (mkFile(&dir, id_trng, "trng", multi_port) < 0) {
 		return -1;
 	}
-#endif
```
Comment (Darchiv) [inline multi/imxrt-multi/imxrt-multi.c:422]: `id_trng` msg is only handled if `#ifndef __CPU_IMXRT117X`, so there is probably no point in creating random/trng devices

File: multi/imxrt-multi/i2c.c [hunk 3]
```diff
@@ -1,84 +1,656 @@
...
+	scfgr1 = 73, scfgr2, samr = 80, sasr = 84, star, stdr = 88, srdr = 92 };
+
+enum { cmd_txdata = 0, cmd_rxdata = (1 << 8), cmd_stop = (2 << 8), cmd_start = (4 << 8) };
+
+enum { i2c_stateBusy = 0, i2c_stateReady = 1 };
+
+/* clang-format on */
+
+
+struct periph_info {
```
Comment (Darchiv) [inline multi/imxrt-multi/i2c.c:96]: `static`?

File: multi/imxrt-multi/i2c.c [hunk 4]
```diff
@@ -1,84 +1,656 @@
...
+	}
+
+	mutexLock(i2c_common[pos].mutex);
+
+	TRACE();
+	volatile uint32_t *base = i2c_common[pos].base;
+
+	/* Reset before configuring */
+	*(base + mcr) = 1u << 1u;
+	*(base + mcr) = 0u;
```
Comment (Darchiv) [inline multi/imxrt-multi/i2c.c:181]: Maybe a memory barrier before and after this write?

File: multi/imxrt-multi/imxrt-multi.c (hunk 2 above)
Comment (jmaksymowicz) [reply multi/imxrt-multi/imxrt-multi.c:422]: It is still behind `TRNG` ifdef so there should be no devices, but I added `#ifndef __CPU_IMXRT117X` to be sure.

File: multi/imxrt-multi/i2c.c (hunk 4 above)
Comment (jmaksymowicz) [reply multi/imxrt-multi/i2c.c:182]: Done

File: multi/imxrt-multi/i2c.c (hunk 3 above)
Comment (jmaksymowicz) [reply multi/imxrt-multi/i2c.c]: Done

**Review by Darchiv — APPROVED** [review]

**Pull request merged**